### ISSUE TITLE
[Question] How to change client's websocket header ?

### ISSUE THREAD

**Issue by klize** [miguelgrinberg/python-socketio#384]
I have two questions
* Is there anyway to change the header after connected ? 
* or Do I have to take the 1) solution ?

I am building some network. In my system, client processes get their own process id to use it as a ID that server can distinguish clients on websocket connection. Moreover the ids are passed as a parameter in each client like this: `sio.connect(header={'cid':os.getpid()})` 

But If there exist two or more client *desktops*, some of clients in different desktops get the same os.getpid() often. So I've been searching , so called, 'PC Identifier' something like MAC or IP.
But MAC or IP can't be the identifier of a desktop. 

1) So I figure it out that I can solve my problems by giving an order to connected clients (like chat room examples), so clients can be distinguishable *after* connection. But then, I always have to send them to the server by including them into the json data when emitting. 

The reason i am not using session id is that unexpected disconnection and reconnection happens and I am not sure sockets are never closed and not reopened before explicit disconnection. 
In short, sid is not stable to use as an ID.

**Comment by miguelgrinberg** [comment]
What do you mean by "header" in this context?

Normally you'd want clients to authenticate, so they'll have to have a username/password or maybe a token assigned to them that they can pass on the connection request. Something similar to your `cid`, but with some security added to it to prevent unauthorized access. If you implement standard authentication then once the client passes the username/password or token you know who they are, and you can write any user data into their user session, which will be available to you every time they send a message to the server.

**Comment by klize** [comment]
'''python
if __name__='__main__':
    ....
    id = '_'.join([ 'client', str(os.getpid()) ])
    sio.connect(host, header={'cid'=id}, transport=['websocket' ])
'''

server.py
'''python
def connect():
    cid=request.header.get('cid')
    clients[cid]= ...
'''
i wanted to use header as a identifier from what client a message is sent. when i emit on client side then the headet contained in the client's sending packet,  server can distinguish what client sent the message. 

The authentication support (username/passwd) as you said above can identify them even any two clients have the same ids e.g.  'client_1234' of A computer and 'client_1234' of B computer? If not possible, i 'd better to find other solutions.
As i am turning on the client processes by for loop in order to do some automated jobs, i can 't give manually on every connection attempts. But this is messed up since A computher generates process 1234 and so does B computer also simultaneously.

**Comment by miguelgrinberg** [comment]
I still don't understand what is the problem that you are having specifically with this package.

If you need unique IDs, why don't you concatenate the computer's IP address with the process ID?

**Comment by klize** [comment]
I just wanted to change header used like below after connection is established
```python
sio.connect(host, header={...}, transport=['websocket'])
```

And I was misunderstanding about usage of header.
Thank you for trying to answer to me. 
IP can be duplicated when i think about all computers over the world, but it can be used in this situation. We can close this.
